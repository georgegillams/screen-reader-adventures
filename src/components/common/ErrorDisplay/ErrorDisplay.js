import React from 'react';
import PropTypes from 'prop-types';
import { Error } from 'gg-components/Error';
import { Paragraph } from 'gg-components/Paragraph';

const ErrorDisplay = props => {
  const { error, message, children, ...rest } = props;

  if (!error) {
    return null;
  }

  const showMessage = !!message;
  const showError = error && error.errorMessage;
  const showChildren = !!children;

  return (
    <Paragraph {...rest}>
      <Error>
        {showMessage && message}
        {showMessage && showError && <br />}
        {showError && error.errorMessage}
        {showChildren && (
          <>
            <br />
            {children}
          </>
        )}
      </Error>
    </Paragraph>
  );
};

ErrorDisplay.propTypes = {
  message: PropTypes.string,
  error: PropTypes.shape({
    errorMessage: PropTypes.string,
  }),
  children: PropTypes.node,
};

ErrorDisplay.defaultProps = {
  message: null,
  error: false,
  children: null,
};

export default ErrorDisplay;
