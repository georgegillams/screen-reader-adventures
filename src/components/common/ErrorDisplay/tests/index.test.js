import React from 'react';

import { render } from '@testing-library/react';

import ErrorDisplay from '..';

describe('<ErrorDisplay  />', () => {
  it('Should render correctly', () => {
    const { container } = render(<ErrorDisplay />);

    expect(container).toMatchSnapshot();
  });

  it('Should render correctly with message', () => {
    const { container } = render(<ErrorDisplay message="Test message" />);

    expect(container).toMatchSnapshot();
  });

  it('Should render with empty error object and message', () => {
    const { container } = render(<ErrorDisplay message="Test message" error={{}} />);

    expect(container).toMatchSnapshot();
  });

  it('Should render with well-defined error object and message', () => {
    const { container } = render(
      <ErrorDisplay message="Test message" error={{ errorMessage: 'Test error message' }} />
    );

    expect(container).toMatchSnapshot();
  });

  it('Should render with empty error object and no message', () => {
    const { container } = render(<ErrorDisplay error={{}} />);

    expect(container).toMatchSnapshot();
  });

  it('Should render with well-defined error object and no message', () => {
    const { container } = render(<ErrorDisplay error={{ errorMessage: 'Test error message' }} />);

    expect(container).toMatchSnapshot();
  });

  it('Should render correctly with children', () => {
    const { container } = render(<ErrorDisplay message="Test message">TEST CHILD</ErrorDisplay>);

    expect(container).toMatchSnapshot();
  });

  it('Should render correctly with custom className', () => {
    const { container } = render(<ErrorDisplay message="Test message" className={'test-custom-classname'} />);

    expect(container).toMatchSnapshot();
  });
});
