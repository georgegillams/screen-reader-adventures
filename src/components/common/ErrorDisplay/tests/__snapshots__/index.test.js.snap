// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorDisplay  /> Should render correctly 1`] = `<div />`;

exports[`<ErrorDisplay  /> Should render correctly with children 1`] = `<div />`;

exports[`<ErrorDisplay  /> Should render correctly with custom className 1`] = `<div />`;

exports[`<ErrorDisplay  /> Should render correctly with message 1`] = `<div />`;

exports[`<ErrorDisplay  /> Should render with empty error object and message 1`] = `
<div>
  <span
    class="paragraph"
  >
    <span
      class="error"
    >
      Test message
    </span>
  </span>
</div>
`;

exports[`<ErrorDisplay  /> Should render with empty error object and no message 1`] = `
<div>
  <span
    class="paragraph"
  >
    <span
      class="error"
    />
  </span>
</div>
`;

exports[`<ErrorDisplay  /> Should render with well-defined error object and message 1`] = `
<div>
  <span
    class="paragraph"
  >
    <span
      class="error"
    >
      Test message
      <br />
      Test error message
    </span>
  </span>
</div>
`;

exports[`<ErrorDisplay  /> Should render with well-defined error object and no message 1`] = `
<div>
  <span
    class="paragraph"
  >
    <span
      class="error"
    >
      Test error message
    </span>
  </span>
</div>
`;
